src/Algebra/Graph/Relation/InternalDerived.hs

```diff
@@ -45,48 +48,167 @@ instance Ord a => Eq (ReflexiveRelation a) where
 instance (Ord a, Show a) => Show (ReflexiveRelation a) where
     show = show . reflexiveClosure . fromReflexive
 
-instance Ord a => Graph (ReflexiveRelation a) where
-    type Vertex (ReflexiveRelation a) = a
-    empty       = ReflexiveRelation empty
-    vertex      = ReflexiveRelation . vertex
-    overlay x y = ReflexiveRelation $ fromReflexive x `overlay` fromReflexive y
-    connect x y = ReflexiveRelation $ fromReflexive x `connect` fromReflexive y
-
-instance Ord a => Reflexive (ReflexiveRelation a)
-
--- TODO: Optimise the implementation by caching the results of symmetric closure.
 {-|  The 'SymmetricRelation' data type represents a /symmetric binary relation/
 over a set of elements. Symmetric relations satisfy all laws of the
-'Undirected' type class and, in particular, the
+'Undirected' type class and, in addition, the
 commutativity of connect:
 
 @'connect' x y == 'connect' y x@
 
 The 'Show' instance produces symmetrically closed expressions:
 
 @show (1     :: SymmetricRelation Int) == "vertex 1"
-show (1 * 2 :: SymmetricRelation Int) == "edges [(1,2),(2,1)]"@
+show (1 * 2 :: SymmetricRelation Int) == "edge 1 2"@
+
+The total order on graphs is defined using /size-lexicographic/ comparison:
+
+* Compare the number of vertices. In case of a tie, continue.
+* Compare the sets of vertices. In case of a tie, continue.
+* Compare the number of edges. In case of a tie, continue.
+* Compare the sets of edges.
+
+Here are a few examples:
+
+@'vertex' 1 < 'vertex' 2
+'vertex' 3 < 'Algebra.Graph.Relation.Symmetric.edge' 1 2
+'vertex' 1 < 'Algebra.Graph.Relation.Symmetric.edge' 1 1
+'Algebra.Graph.Relation.Symmetric.edge' 1 2 = 'Algebra.Graph.Relation.Symmetric.edge' 2 1
+'Algebra.Graph.Relation.Symmetric.edge' 1 1 < 'Algebra.Graph.Relation.Symmetric.edge' 1 2
+'Algebra.Graph.Relation.Symmetric.edge' 1 2 > 'Algebra.Graph.Relation.Symmetric.edge' 1 1 + 'Algebra.Graph.Relation.Symmetric.edge' 2 2
```

Comment:
Is this really true? I don't think so. All these examples should be tested to make sure they pass.

Reply:
It's not, in my tests it has a '<', don't know what happened there.. sorry!